closed/test/jtreg-ext/jtreg/openj9/CoreDumpTimeoutHandler.java

```diff
@@ -113,16 +113,28 @@ private void runJcmd(long pid) throws InterruptedException {
                 return;
             }
 
-            ProcessBuilder pb = new ProcessBuilder(jcmd.getAbsolutePath(), Long.toString(pid), "Dump.system");
-            pb.redirectErrorStream(true);
+            ProcessBuilder pbcore = new ProcessBuilder(jcmd.getAbsolutePath(), Long.toString(pid), "Dump.system");
+            pbcore.redirectErrorStream(true);
 
-            Process p = pb.start();
-            try (BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+            Process pcore = pbcore.start();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(pcore.getInputStream()))) {
                 String line;
                 while ((line = reader.readLine()) != null) {
                     log.println(line);
                 }
-                p.waitFor();
+                pcore.waitFor();
+            }
+ 
+            ProcessBuilder pbjavacore = new ProcessBuilder(jcmd.getAbsolutePath(), Long.toString(pid), "Dump.java");
+            pbjavacore.redirectErrorStream(true);
+
+            Process pjavacore = pbjavacore.start();
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(pjavacore.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    log.println(line);
+                }
+                pjavacore.waitFor();
             }
```

Comment:
Perhaps this repetition should be extracted into a helper method?

Reply:
Updated.